src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -59,12 +197,37 @@ protected virtual void Dispose(bool disposing)
             {
                 if (disposing)
                 {
-                    _cancellationTokenSource.Dispose();
+                    _startupLoopTokenSource?.Dispose();
                 }
                 _disposed = true;
             }
         }
 
         public void Dispose() => Dispose(true);
     }
+
+    public class CustomFactory : ILoggerFactory
```

Comment:
Why is this type defined in this file? If it is a temporary type, recommend moving to where it is used and making internal/private if possible?